scripts/docs-deploy/deploy-to-site.ts

```diff
@@ -0,0 +1,57 @@
+import {$} from 'zx';
+import {SiteTarget} from './utils';
+
+interface Deployment {
+  projectId: string;
+  site: SiteTarget;
+}
+
+/** Interface describing a production deployment. */
+export interface ProductionDeployment extends Deployment {
+  description: string;
+}
+
+/** Interface describing a temporary preview deployment. */
+export interface PreviewDeployment extends Deployment {
+  channelId: string;
+  expires: string;
+}
+
+/** Type describing a Firebase deployment. */
+export type DeploymentInfo = ProductionDeployment | PreviewDeployment;
+
+/**
+ * Deploys the docs site at the specified directory to Firebase with respect
+ * to the deployment information provided.
+ *
+ * The deployment info either describes the production deployment information,
+ * or a preview temporary deployment that will expire automatically.
+ */
+export async function deployToSite(projectPath: string, token: string, info: DeploymentInfo) {
+  // Note: Running yarn in silent move to avoid printing of tokens.
+  const firebase = async (...cmd: string[]) =>
+    await $`yarn -s --cwd ${projectPath} firebase --non-interactive --token "${token}" ${cmd}`;
```

Comment:
nit: this function seems like something that we could pull out into a helper, rather than keeping it inline.

Reply:
I've removed the token completely, so this is just a local helper now to avoid needing to set the CWD.